impl/opts.go

```
@@ -1,13 +1,23 @@
 package impl
 
-import "time"
+import (
+	"time"
+
+	"github.com/handcoding-labs/redis-stream-client-go/configs"
+)
 
 type RecoverableRedisOption func(*RecoverableRedisStreamClient)
 
 // WithLBSIdleTime sets the time after which a message is considered idle and will be recovered
 func WithLBSIdleTime(idleTime time.Duration) RecoverableRedisOption {
 	return func(r *RecoverableRedisStreamClient) {
-		r.lbsIdleTime = idleTime
+		// idleTime must be greater than 2 * heartbeat interval at least
+		if idleTime != 0 && idleTime > (2*r.hbInterval) {
+			r.lbsIdleTime = idleTime
+		}
+
+		// if the value is not valid, set it to the default
+		r.lbsIdleTime = configs.DefaultLBSIdleTime
```

This needs to be in an `else`, otherwise we'd always be setting it to the default idle time.

Also: In the event the user specifies a custom value, should we silently overwrite their setting to the default if we deem their value to not be suitable? Shouldn't we return an error?